Source/Core/VideoBackends/Vulkan/VKPerfQuery.cpp

```diff
@@ -126,7 +126,9 @@ u32 PerfQuery::GetQueryResult(PerfQueryType type)
 
 void PerfQuery::FlushResults()
 {
-  PartialFlush(true);
+  if (!IsFlushed())
```

Comment:
Would be better if this change got put into the first commit

Reply:
Also, why was this changed from a `while`?

Reply:
It was changed from a `while` because `IsFlushed` is always true after one iteration.

`PartialFlush(true)` executes the current command buffer and immediately waits for that to finish. Therefore, all GPU queries are also done after that.